selfdrive/sensord/tests/test_sensord.py

```diff
@@ -55,24 +115,162 @@ def setUpClass(cls):
-    sensor_events = messaging.sub_sock("sensorEvents", timeout=0.1)
-
-    start_time_sec = time.time()
-    events = []
-    while time.time() - start_time_sec < TEST_TIMESPAN:
-      events += messaging.drain_sock(sensor_events)
-      time.sleep(0.01)
+    # verify correct sensors configuration
+    events = read_sensor_events(10)
 
     seen = set()
     for event in events:
       for measurement in event.sensorEvents:
-        # Filter out unset events
+        # filter unset events (bmx magn)
         if measurement.version == 0:
           continue
         seen.add((str(measurement.source), measurement.which()))
 
     self.assertIn(seen, SENSOR_CONFIGURATIONS)
 
+  @with_processes(['sensord'])
+  def test_lsm6ds3_100Hz(self):
+    # verify measurements are sampled and published at a 100Hz rate
+    events = read_sensor_events(3) # 3sec (about 300 measurements)
+
+    data_points = set()
+    for event in events:
+      for measurement in event.sensorEvents:
+
+        # skip lsm6ds3 temperature measurements
+        if measurement.which() == 'temperature':
+          continue
+
+        if str(measurement.source).startswith("lsm6ds3"):
+          data_points.add(measurement.timestamp)
+
+    assert len(data_points) != 0, "No lsm6ds3 sensor events"
+
+    data_list = list(data_points)
+    data_list.sort()
+    tdiffs = np.diff(data_list)
+
+    high_delay_diffs = set(filter(lambda d: d >= 10*10**6, tdiffs))
+    assert len(high_delay_diffs) < 10, f"Too many high delay packages: {high_delay_diffs}"
+
+    avg_diff = sum(tdiffs)/len(tdiffs)
+    assert avg_diff > 9.6*10**6, f"avg difference {avg_diff}, below threshold"
+
+    stddev = np.std(tdiffs)
+    assert stddev < 1.5*10**6, f"Standard-dev to big {stddev}"
+
+  @with_processes(['sensord'])
+  def test_events_check(self):
+    # verify if all sensors produce events
+    events = read_sensor_events(3)
+
+    sensor_events = dict()
+    for event in events:
+      for measurement in event.sensorEvents:
+
+        # filter unset events (bmx magn)
+        if measurement.version == 0:
+          continue
+
+        if measurement.type in sensor_events:
+          sensor_events[measurement.type] += 1
+        else:
+          sensor_events[measurement.type] = 1
+
+    for s in sensor_events:
+      err_msg = f"Sensor {s}: 200 < {sensor_events[s]}"
+      assert sensor_events[s] > 200, err_msg
+
+  @with_processes(['sensord'])
+  def test_logmonottime_timestamp_diff(self):
+    # ensure diff between the message logMonotime and sample timestamp is small
+    events = read_sensor_events(3)
+
+    tdiffs = list()
+    for event in events:
+      for measurement in event.sensorEvents:
+
+        # filter unset events (bmx magn)
+        if measurement.version == 0:
+          continue
+
+        # negative values might occur, as non interrupt packages created
+        tdiffs.append(abs(event.logMonoTime - measurement.timestamp))
+        # before the sensor is read
```

Comment:
let's explicitly check the interrupts ones here and make sure that the logMonoTime is after the sample ts. it's ok to just hardcode the list

Reply:
oh I see that's a different test case